openquake/hmtk/seismicity/catalogue.py

```diff
@@ -107,8 +115,37 @@ def get_number_events(self):
     def add_event(self):
         raise NotImplementedError
 
-    def write_catalogue(self, output_file, filetype):
-        raise NotImplementedError
+    def write_catalogue(self, output_file, key_list=SORTED_ATTRIBUTE_LIST):
+        """
+        Writes the catalogue to file using HTMK format (CSV).
+
+        :param output_file:
+            Name of the output file
+        :param key_list:
+            Optional list of attribute keys to be exported
+        """
+
+        with open(output_file, 'wt') as of:
+            writer = csv.DictWriter(of, fieldnames=key_list)
```

Comment:
BTW, since 'rt' and 'wt' are the same as 'r' and 'w', nobody uses the long form.